### PR TITLE
Remove AnimationDirection

### PR DESCRIPTION
**Pull request by abarth**
This concept is now private to AnimationController. All the clients actually
want the AnimationStatus.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: examples/material_gallery/lib/demo/progress_indicator_demo.dart [hunk 1]
```diff
@@ -14,39 +14,44 @@ class _ProgressIndicatorDemoState extends State<ProgressIndicatorDemo> {
...
+    )..forward();
 
     animation = new CurvedAnimation(
       parent: controller,
       curve: new Interval(0.0, 0.9, curve: Curves.ease),
       reverseCurve: Curves.ease
     )..addStatusListener((AnimationStatus status) {
-      if (status == AnimationStatus.dismissed || status == AnimationStatus.completed)
-        reverseValueAnimationDirection();
+      if (status == AnimationStatus.dismissed)
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/progress_indicator_demo.dart]: this is probably best as a switch with an explicitly empty default case

File: packages/flutter/lib/src/material/tabs.dart [hunk 2]
```diff
@@ -906,16 +905,11 @@ class _TabBarViewState extends PageableListState<TabBarView> implements TabBarSe
...
     } else {
       _updateItemsFromChildren(previousSelectedIndex, selectedIndex);
-      _scrollDirection = AnimationDirection.forward;
-    }
-
-    if (_scrollDirection == AnimationDirection.forward)
       scrollTo(animation.value);
-    else
-      scrollTo(1.0 - animation.value);
+    }
```
Comment (Hixie) [inline packages/flutter/lib/src/material/tabs.dart]: cc @HansMuller for this bit

File: packages/flutter/lib/src/widgets/heroes.dart [hunk 3]
```diff
@@ -479,7 +479,7 @@ class HeroController extends NavigatorObserver {
 
   void _addHeroToOverlay(Widget hero, Object tag, OverlayState overlay) {
     OverlayEntry entry = new OverlayEntry(builder: (_) => hero);
-    if (_animation.direction == AnimationDirection.forward)
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/heroes.dart]: assert that the _animation.status isn't complete or dismissed

**Comment by Hixie** [comment]
lgtm

**Pull request merged**